Fix Repairable crash

reaperrr opened the pull request:
`MoveAdjacentTo` is a `Mobile`-only activity, and `Repairable.AfterReachActivities` was generally written with ground actors in mind.
Reusing the "Enter" logic in `Aircraft` is both simpler and safer (regarding regression risk) than trying to adapt `Repairable`.

Fixes #15820.
Closes #13975.

reaperrr committed:
Fix Repairable crash

MoveAdjacentTo is a Mobile-only activity.

pchote reviewed (dismissed):
This indeed fixes the crash, but helicopters now repair in the air above the helipad/service depot without landing.

Repro case: build an apache/orca in TD, damage them, then use the repair order generator on them.

reaperrr committed:
Fix aircraft being repaired mid-air

Repairable was originally written for ground actors,
so it's both safer and much easier to just handle this in Aircraft directly.

pchote commented:
BTW, when did we (accidentally?) fix the repair cursor for aircraft?

#13975 is still open, and I don't see a mention in the changelog.

reaperrr commented:
If one of my PRs fixed it, it was by accident.

pchote commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/d94fc9a51c5b87dca38d8f6b0d356e01d17fc2c0

> If one of my PRs fixed it, it was by accident.

it was fixed by #15543, which is now reflected in the changelog
